### PR TITLE
Feature: Add support for REGEX for test subcommand's filter flag

### PR DESCRIPTION
**Pull request by sebastienfilion**
Currently, the documentation makes it sound like the test subcommand's filter flag could accept some kind of pattern matching value like a glob or a REGEX, although the function `createFilterFn` accepts a regex as an argument, there's no way to pass an actual REGEX value from the CLI.

This commit makes it possible to pass a string that could be cast as REGEX when string = `^/.*/$`.

With this change, a user can use the filter flag as follow:

```shell
deno test --filter "/test-.+/"
```

I've also tested that `\` get escaped properly, on MacOS at least, and this is also a valid flag:

```shell
deno test --filter "/test-\d+/"
```

Reference: https://github.com/denoland/deno/pull/6323

### CODE REVIEW

**Comment by wongjiahau** [comment]
@sebastienfilion Can you add documentation about this feature?

**Comment by sebastienfilion** [comment]
@wongjiahau yes, I wanted to wait for https://github.com/denoland/deno/pull/6323#pullrequestreview-433139186 to get merged in. 😊

File: cli/js/testing.ts [hunk 1]
```diff
@@ -307,6 +307,9 @@ function createFilterFn(
     if (filter) {
       if (filter instanceof RegExp) {
         passes = passes && filter.test(def.name);
+      } else if (/^\/.*\/$/.test(filter)) {
```
Comment (ry) [inline cli/js/testing.ts:310]: `filter.startsWith('/') && filter.endsWith('/')` is probably faster

File: cli/js/testing.ts [hunk 2]
```diff
@@ -307,6 +307,9 @@ function createFilterFn(
...
+        const filterAsRegex = new RegExp(filter.replace(/^\/|\/$/g, ""));
```
Comment (ry) [inline cli/js/testing.ts:311]: `filter.slice(1, filter.length - 1)` is probably better than `filter.replace(/^\/|\/$/g, "")`

File: cli/tests/unit/filter_function_test.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    {
+      fn(): void {},
+      name: "other-test",
+    },
+  ];
+
+  const filteredTests = tests.filter(filterFn);
+
+  assertEquals(filteredTests.length, 2);
+});
```
Comment (ry) [inline cli/tests/unit/filter_function_test.ts:58]: Nice tests! Can you reduce the vertical white space by deleting some of these empty lines?

File: cli/js/testing.ts (hunk 2 above)
Comment (sebastienfilion) [reply cli/js/testing.ts]: Thanks @ry I guess I got really excited by REGEX 😅

**Comment by ry** [comment]
LGTM - but holding for 1.2.0

**Pull request merged**